packages/matter.js/src/fabric/FabricManager.ts

```diff
@@ -6,74 +6,96 @@
 
 import { InternalError, MatterError, MatterFlowError } from "../common/MatterError.js";
 import { FabricIndex } from "../datatype/FabricIndex.js";
-import { NodeId } from "../datatype/NodeId.js";
 import { StorageContext } from "../storage/StorageContext.js";
 import { ByteArray } from "../util/ByteArray.js";
+import { Observable } from "../util/Observable.js";
 import { Fabric, FabricJsonObject } from "./Fabric.js";
 
 /** Specific Error for when a fabric is not found. */
 export class FabricNotFoundError extends MatterError {}
 export class FabricTableFullError extends MatterError {}
 
+export enum FabricAction {
+    Added,
+    Removed,
+    Updated,
+}
+
 export class FabricManager {
-    private nextFabricIndex = 1;
-    private readonly fabrics = new Map<FabricIndex, Fabric>();
+    #nextFabricIndex = 1;
+    readonly #fabrics = new Map<FabricIndex, Fabric>();
+    #initializationDone = false;
+    #fabricStorage: StorageContext;
+    #events = {
+        added: Observable<[fabric: Fabric]>(),
+        updated: Observable<[fabric: Fabric]>(),
+        removed: Observable<[fabric: Fabric]>(),
```

Comment:
Nbd but maybe use "deleted" to match conventions

Reply:
done too